src/aead/aead.rs

```diff
@@ -396,38 +425,29 @@ mod tests {
             ];
 
             for in_prefix in IN_PREFIXES.iter() {
-                let max_overhead_len = aead_alg.max_overhead_len();
-                let mut s_in_out = plaintext.clone();
-                for _ in 0..max_overhead_len {
-                    s_in_out.push(0);
-                }
-                let s_key = aead::SealingKey::new(aead_alg, &key_bytes).unwrap();
-                let s_result = aead::seal_in_place(&s_key, &nonce,
-                                                   &mut s_in_out[..],
-                                                   max_overhead_len, &ad);
-
-                let mut o_in_out = Vec::from(*in_prefix);
-                o_in_out.extend_from_slice(&ct[..]);
-                let o_key = aead::OpeningKey::new(aead_alg, &key_bytes).unwrap();
-                let o_result = aead::open_in_place(&o_key, &nonce,
-                                                   in_prefix.len(),
-                                                   &mut o_in_out[..], &ad);
-
-                match error {
-                    None => {
-                        assert_eq!(Ok(ct.len()), s_result);
-                        assert_eq!(&ct[..], &s_in_out[..ct.len()]);
-                        assert_eq!(Ok(plaintext.len()), o_result);
-                        assert_eq!(&plaintext[..], &o_in_out[..plaintext.len()]);
-                    },
-                    Some(ref error) if error == "WRONG_NONCE_LENGTH" => {
-                        assert_eq!(Err(()), s_result);
-                        assert_eq!(Err(()), o_result);
-                    },
-                    Some(error) => {
-                        unreachable!("Unexpected error test case: {}", error);
-                    }
-                };
+                test_aead_inner(aead_alg, &key_bytes, &nonce, &plaintext,
+                                &ad, &ct, &error, in_prefix)
+            }
+        });
+    }
+
+    fn test_aead_slow(aead_alg: &'static aead::Algorithm, file_path: &str) {
+        file_test::run(file_path, |section, test_case| {
+            assert_eq!(section, "");
+            let key_bytes = test_case.consume_bytes("KEY");
+            let nonce = test_case.consume_bytes("NONCE");
+            let plaintext = test_case.consume_bytes("IN");
+            let ad = test_case.consume_bytes("AD");
+            let mut ct = test_case.consume_bytes("CT");
+            let tag = test_case.consume_bytes("TAG");
+            let error = test_case.consume_optional_string("FAILS");
+
+            ct.extend(tag);
+
+            for in_prefix_len in 0..4096 {
+                let in_prefix = &vec![123; in_prefix_len][..];
```

Comment:
Perhaps the test would run faster if, instead of using `vec!`, we did something like `static PREFIX: [u8; 4096] = [123; 4096]` (outside the loop) and then `let in_prefix = PREFIX[..in_prefix_len]` or similar?